diff --git a/tactics/auto.ml b/tactics/auto.ml
index d03c527..ecfa77d 100644
--- a/tactics/auto.ml
+++ b/tactics/auto.ml
@@ -1127,7 +1127,7 @@ let search = search_gen 0

 let default_search_depth = ref 5

-let delta_auto mod_delta n lems dbnames gl =
+let delta_auto ?(use_core_db=true) mod_delta n lems dbnames gl =
   let db_list =
     List.map
       (fun x ->
@@ -1135,11 +1135,11 @@ let delta_auto mod_delta n lems dbnames gl =
 	   searchtable_map x
 	 with Not_found ->
 	   error_no_such_hint_database x)
-      ("core"::dbnames)
+      (if use_core_db then "core"::dbnames else dbnames)
   in
   tclTRY (search n mod_delta db_list (make_local_hint_db false lems gl)) gl

-let auto = delta_auto false
+let auto ?(use_core_db=true) = delta_auto ~use_core_db false

 let new_auto = delta_auto true

diff --git a/tactics/auto.mli b/tactics/auto.mli
index cd8808b..54131aa 100644
--- a/tactics/auto.mli
+++ b/tactics/auto.mli
@@ -192,7 +192,7 @@ val conclPattern : constr -> constr_pattern option -> Tacexpr.glob_tactic_expr -

 (** The Auto tactic *)

-val auto : int -> open_constr list -> hint_db_name list -> tactic
+val auto : ?use_core_db:bool -> int -> open_constr list -> hint_db_name list -> tactic

 (** Auto with more delta. *)

